app/models/host/discovered.rb

```diff
@@ -13,8 +12,8 @@ class Host::Discovered < ::Host::Base
 
   scoped_search :on => :name, :complete_value => true, :default_order => true
   scoped_search :on => :last_report, :complete_value => true
-  scoped_search :on => :ip, :complete_value => true
-  scoped_search :on => :mac, :complete_value => true
+  scoped_search :in => :primary_interface, :on => :ip, :complete_value => true
+  scoped_search :in => :primary_interface, :on => :mac, :complete_value => true
```

Comment:
we don't need the subnet because of:  `has_one :subnet, :through => :primary_interface` in host/base.rb
we only show the ip and mac from primary interface in the search results so wouldn't that be a little confusing to search on all interfaces?